Vector_graphics_on_surfaces/include/CGAL/Vector_graphics_on_surfaces/locally_shortest_path.h

```diff
@@ -0,0 +1,4814 @@
+  const Dual_geodesic_solver<typename K::FT>* solver_ptr=&solver;
+  Dual_geodesic_solver<typename K::FT> local_solver;
+  if (solver.graph.empty())
+  {
+    solver_ptr = &local_solver;
+    init_geodesic_dual_solver(local_solver, tmesh);
+  }
+#endif
+
+#ifdef CGAL_DEBUG_BSURF
+  std::ofstream debug_cp("/tmp/control_points.xyz");
+  std::ofstream debug_ep("/tmp/end_points.xyz");
+  debug_cp << std::setprecision(17);
+  debug_ep << std::setprecision(17);
+#endif
+  for (std::size_t i=0; i<n; ++i)
+  {
+    Bezier_segment<TriangleMesh, FT> control_loc;
+    for (int k=0;k<4; ++k)
+    {
+      control_loc[k] = straightest_geodesic<K>(center,directions[i][k],lengths[i][k],tmesh).back();
+    }
+
+    #ifdef CGAL_DEBUG_BSURF
+      debug_ep << PMP::construct_point(control_loc[0], tmesh) << "\n";
+      debug_ep << PMP::construct_point(control_loc[3], tmesh) << "\n";
+      debug_cp << PMP::construct_point(control_loc[1], tmesh) << "\n";
+      debug_cp << PMP::construct_point(control_loc[2], tmesh) << "\n";
+    #endif
+
+    std::vector<PMP::Face_location<TriangleMesh, FT>> bezier =
+      recursive_de_Casteljau(tmesh, control_loc, num_subdiv
+#ifndef CGAL_BSURF_USE_DIJKSTRA_SP
+                            , *solver_ptr
+#endif
+                            );
+
+    result[i].reserve(bezier.size());
+    result[i].push_back(bezier[0]);
+    for(std::size_t b=0; b<bezier.size()-1; ++b)
+    {
+      const PMP::Face_location<TriangleMesh, FT>& loc = bezier[b];
+      const PMP::Face_location<TriangleMesh, FT>& loc1 = bezier[b+1];
+
+      // connect the two face locations with shortest path is they are in different faces
+      if (loc.first!=loc1.first)
+      {
+        std::vector<PMP::Edge_location<TriangleMesh, FT>> edge_locations;
+        locally_shortest_path<FT>(loc, loc1, tmesh, edge_locations, solver);
+        result[i].reserve(result[i].size()+edge_locations.size());
+        for (const PMP::Edge_location<TriangleMesh, FT>& e : edge_locations)
+          result[i].push_back(PMP::to_face_location(e, tmesh));
+      }
+      result[i].push_back(loc1);
+    }
+  }
+
+  return result;
+}
+
+
+/*!
+ * \ingroup VGSFunctions
+ * computes a path representing a Bézier polyline (a sequence of Bézier curves having an identical control points,
+ * that is the fourth control point of the nth curve is the first control point of the (n+1)th curve).
+ * Control points are defined by the endpoints of straightest geodesic curves starting from `center` along given directions and distances.
+ * The iterative de Casteljau subdivision algorithm is applied to create more control points that are then connected with locally shortest paths.
+ * The output path is such that for two consecutive face locations, there must exist a face in `tmesh` containing the two corresponding points.
+ * The first portion of the curve is defined by the 4 first values in `directions` and `lengths`. The second portion is defined by the 4'th value
```

Comment:
Maybe "by the next 4 values".    Also I am not sure if instead of 4 you should write four